### PR TITLE
Change default behavior of autotor onions to Version 3 onions.

### PR DESCRIPTION
**Pull request by Saibato**
Since V3 onions are now supported on most platforms and distros we can default to V3 on autotor service onions.

### CODE REVIEW

File: connectd/tor_autoservice.c [hunk 1]
```diff
@@ -73,17 +73,41 @@ static void discard_remaining_response(struct rbuf *rbuf)
...
+/* Now that V3 is out of Beta default to V3 autoservice onions if version is above 0.4 
+*/
+	tor_send_cmd(rbuf, "PROTOCOLINFO 1");
+
+	while ((line = tor_response_line(rbuf)) != NULL) {
+
+		if (!strstarts(line, "VERSION Tor="))
+			continue;
+
+    if (use_v3_autotor)
```
Comment (ZmnSCPxj) [inline connectd/tor_autoservice.c]: Inconsistent indent.

File: connectd/tor_autoservice.c [hunk 2]
```diff
@@ -73,17 +73,41 @@ static void discard_remaining_response(struct rbuf *rbuf)
...
+		if(strstr(line, "\"0.0") ||
```
Comment (ZmnSCPxj) [inline connectd/tor_autoservice.c]: Also these and succeeding lines.

File: connectd/tor_autoservice.c [hunk 3]
```diff
@@ -73,17 +73,41 @@ static void discard_remaining_response(struct rbuf *rbuf)
...
+			strstr(line, "\"0.1") ||
+			strstr(line, "\"0.2") ||
+			strstr(line, "\"0.3"))
+					use_v3_autotor = false;
```
Comment (ZmnSCPxj) [inline connectd/tor_autoservice.c]: Probably deserves a `status_unusual` log message as well?

File: connectd/tor_autoservice.c [hunk 4]
```diff
@@ -224,8 +249,10 @@ struct wireaddr *tor_autoservice(const tal_t *ctx,
 	buffer = tal_arr(tmpctx, char, rbuf_good_size(fd));
 	rbuf_init(&rbuf, fd, buffer, tal_count(buffer), buf_resize);
 
+	use_v3_autotor = true;
```
[the remark this answers is not in the document]
Comment (Saibato) [reply connectd/tor_autoservice.c]: Yes thought about that and is WIP:
Therefor for now the var value, the PR for that is quite a hassle as it has to go all trough the message system, or do we have an other way to use globals ?

[the remark this answers is not in the document]
Comment (ZmnSCPxj) [reply connectd/tor_autoservice.c]: Maybe OK to use a global within `connectd` though you still have to pass the flag boolean via the `connectd` startup message.

**Comment by cdecker** [comment]
The first commit breaks bisectability:

```
connectd/tor_autoservice.c:82:/* Now that V3 is out of Beta default to V3 autoservice onions if version is above 0.4 
Extraneous whitespace found
Makefile:298: recipe for target 'check-whitespace/connectd/tor_autoservice.c' failed
make: *** [check-whitespace/connectd/tor_autoservice.c] Error 1
```

If possible I'd like that to be fixed :-)

**Comment by Saibato** [comment]
> Edit: or we could just squash the PR into a single commit if that's easier

I will squash it, I adapted the idea of @ZmnSCPxj to make it configurable so changed the PR to WIP

File: connectd/tor_autoservice.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (Saibato) [reply connectd/tor_autoservice.c]: done :-)

**Pull request merged**